dynamicdns.go

```diff
@@ -292,6 +308,30 @@ func stringListContains(list []string, s string) bool {
 	return false
 }
 
+type IPVersion string
+
+const (
+	IPv4 IPVersion = "ipv4"
+	IPv6 IPVersion = "ipv6"
+)
+
+func (ip IPVersion) IsValid() error {
+	switch ip {
+	case IPv4, IPv6:
+		return nil
+	}
+	return errors.New("Invalid IP version")
+}
+
+func IPVersionsContains(versions []IPVersion, version IPVersion) bool {
```

Comment:
I get why this is useful, but maybe it'd be a little cleaner if we define another type `IPVersions []IPVersion` that has `Contains()` or `Includes()` or `Has()` as a method.

Reply:
Actually since there's only 2 IP versions, what if we don't use a slice at all:

```go
type IPVersions struct {
    ipv4, ipv6 bool
}
```

then have methods to get/set IPv4 and IPv6.

OR, maybe even better yet, we could even do something like what `os.FileMode` does: a simple bitmask so you can specify versions like `IPv4|IPv6`.

I think I like the bitmask approach the most.

Reply:
Cool! I need this feature because i want to disable IPV4 record adding :P

Reply:
Alright, changed to use a struct of bools, using `*bool` so that omission still means enabled (just nicer ergonomics)